### PR TITLE
Simplify the nodes handling method definition

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: src/main/java/org/truffleruby/language/methods/LiteralMethodDefinitionNode.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            if (getDefaultDefineeNode == null) {
+                getDefaultDefineeNode = insert(new GetDefaultDefineeNode());
+            }
+            return getDefaultDefineeNode.getDefaultDefinee(currentMethod, self, declarationContext);
+        } else {
+            return null;
+        }
+    }
+
+}
```
Comment (chrisseaton) [inline src/main/java/org/truffleruby/language/methods/LiteralMethodDefinitionNode.java]: Does Git realise that this is a rename?

Comment (eregon) [reply src/main/java/org/truffleruby/language/methods/LiteralMethodDefinitionNode.java]: It's not a rename, it's logic from 2 different files.

File: src/main/java/org/truffleruby/core/module/ModuleNodes.java [hunk 2]
```diff
@@ -1840,10 +1841,10 @@ public SetMethodVisibilityNode(Visibility visibility) {
             this.visibility = visibility;
         }
 
-        public abstract DynamicObject executeSetMethodVisibility(VirtualFrame frame, DynamicObject module, Object name);
+        public abstract Object executeSetMethodVisibility(VirtualFrame frame, DynamicObject module, Object name);
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/core/module/ModuleNodes.java]: Why change the return type here?

File: src/main/java/org/truffleruby/language/methods/AddMethodNode.java [hunk 3]
```diff
@@ -42,7 +42,7 @@ public AddMethodNode(boolean ignoreNameVisibility) {
         this.ignoreNameVisibility = ignoreNameVisibility;
     }
 
-    public abstract DynamicObject executeAddMethod(DynamicObject module, InternalMethod method, Visibility visibility);
+    public abstract Object executeAddMethod(DynamicObject module, InternalMethod method, Visibility visibility);
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/language/methods/AddMethodNode.java]: Why change the return type here?

File: src/main/java/org/truffleruby/core/module/ModuleNodes.java (hunk 2 above)
Comment (eregon) [reply src/main/java/org/truffleruby/core/module/ModuleNodes.java]: Not needed, indeed. I wanted to make it `void` but I think that doesn't work for some reason.

Comment (eregon) [reply src/main/java/org/truffleruby/core/module/ModuleNodes.java]: The idea is it better expresses that the return value shouldn't be used compared to `DynamicObject`.

Comment (eregon) [reply src/main/java/org/truffleruby/core/module/ModuleNodes.java]: Inheriting RubyBaseNode allows to have `void` execute* methods, so this is fixed now.

**Pull request merged**